### DISCUSSION TITLE
gRPC and Global Exception Mappers

### DISCUSSION THREAD

**Discussion by senthilredhat** [quarkusio/quarkus#26869] [Q&A]
I am looking for a custom exception mapper for gRPC in Quarkus similar to @ServerExceptionMapper for JAX-RS. Any solution around global exception handling in gRPC and Quarkus would be helpful. Thank you.

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment]
We don't have such a feature yet. Can you open an issue and explain what kind of API you have in mind?

**Reply by senthilredhat** [reply]
Sure, I will create an issue with my thoughts. 
-Thank you.

**Comment by elrob** [comment]
I found this after being frustrated that attempts to return custom exceptions and map them to grpc StatusRuntimeException in an interceptor require:
1. Extracting the original exception from the cause of the StatusRuntimeException in the interceptor
2. Putting up with this warning log: https://github.com/quarkusio/quarkus/blob/a4a9c3996009271b15b64c0d12102d979fce7801/extensions/grpc/stubs/src/main/java/io/quarkus/grpc/stubs/ServerCalls.java#L230

My solution was to have my custom exception extend `StatusRuntimeException` with the status and description of my choosing.

It would be great if there was a way (similar to `@ServerExceptionMapper`) which could be utilized map exceptions globally before this call: https://github.com/quarkusio/quarkus/blob/a4a9c3996009271b15b64c0d12102d979fce7801/extensions/grpc/stubs/src/main/java/io/quarkus/grpc/stubs/ServerCalls.java#L219

**Reply by cescoffier** [reply]
Yes, @alesj will look into that for the "new" gRPC server.

**Reply by elrob** [reply]
Thank you!
It's not a huge issue but it feels like I'm conflating things which should be separate concerns when I make a business-logic exception extend the gRPC `StatusRuntimeException`

**Reply by elrob** [reply]
This seems to have been implemented and released with v `2.16.0.Latest` 
PR: https://github.com/quarkusio/quarkus/pull/29419
Issue: https://github.com/quarkusio/quarkus/issues/28609
